modules/multisignatures.js

```diff
@@ -67,171 +74,254 @@ function Multisignatures(cb, scope) {
+ * @param {string} signature.signature - Actual signature
+ * @param {Object} transaction - Corresponding transaction grabbed from transaction pool
+ * @param {function} cb - Callback function
+ * @implements {__private.validateSignature}
+ * @returns {setImmediateCallback} cb, err
+ */
+__private.processSignatureForMultisignatureAccountCreation = (
+	signature,
+	transaction,
+	cb
+) => {
+	// Normalize members of multisignature account from transaction
+	const membersPublicKeys = transaction.asset.multisignature.keysgroup.map(
+		member => member.substring(1) // Remove first character, which is '+'
+	);
 
-		// Find public key
-		let verify = false;
-
-		try {
-			for (
-				let i = 0;
-				i < multisignatureTransaction.asset.multisignature.keysgroup.length &&
-				!verify;
-				i++
-			) {
-				const key = multisignatureTransaction.asset.multisignature.keysgroup[
-					i
-				].substring(1);
-				verify = library.logic.transaction.verifySignature(
-					multisignatureTransaction,
-					key,
-					transaction.signature
-				);
-			}
-		} catch (e) {
-			library.logger.error(e.stack);
-			return setImmediate(cb, 'Failed to verify signature');
-		}
+	// Set empty object as sender, as we don't need sender in case of multisignature registration
+	const sender = {};
 
-		if (!verify) {
-			return setImmediate(cb, 'Failed to verify signature');
-		}
+	return __private.validateSignature(
+		signature,
+		membersPublicKeys,
+		transaction,
+		sender,
+		cb
+	);
+};
 
-		return done(cb);
-	}
+/**
+ * Process signature for transactions that comes from already estabilished multisignature account
+ *
+ * @private
+ * @param {Object} signature - Signature data
+ * @param {string} [signature.publicKey] - Public key of account that created the signature (optional)
+ * @param {string} signature.transactionId - Id of transaction that signature was created for
+ * @param {string} signature.signature - Actual signature
+ * @param {Object} transaction - Corresponding transaction grabbed from transaction pool
+ * @param {function} cb - Callback function
+ * @implements {__private.validateSignature}
+ * @returns {setImmediateCallback} cb, err
+ */
+__private.processSignatureFromMultisignatureAccount = (
+	signature,
+	transaction,
+	cb
+) => {
+	// Get sender account of correscponding transaction
 	modules.accounts.getAccount(
-		{
-			address: multisignatureTransaction.senderId,
-		},
-		(err, account) => {
-			if (err) {
-				return setImmediate(cb, 'Multisignature account not found');
+		{ address: transaction.senderId },
+		(err, sender) => {
+			if (err || !sender) {
+				const message = 'Unable to process signature, account not found';
+				library.logger.error(message, { signature, transaction, error: err });
+				return setImmediate(cb, new Error(message));
 			}
 
-			let verify = false;
-			const multisignatures = account.multisignatures;
+			// Assign members of multisignature account from transaction
+			const membersPublicKeys = sender.multisignatures;
```

Comment:
don't need to account for u_multisingatures?

Reply:
It's not needed. In that case here `sender` have to be already a multisignature account, so need to contains `multisignatures` property (confirmed state). In case of collecting signatures for multisignature account registration transaction - we're grabbing members directly from transaction.